File: frontend/src/lib/tauri-api/deep-links.ts
import type { OpenDeepLinkInput, OpenDeepLinkResult } from '../../types/deep-link';
import { invokeCommand } from './client';

export function openDeepLink(input: OpenDeepLinkInput): Promise<OpenDeepLinkResult> {
  return invokeCommand<OpenDeepLinkResult>('open_deep_link', { input });
}

File: frontend/src-tauri/src/commands/deep_links.rs
use tauri::State;

use crate::models::{OpenDeepLinkInput, OpenDeepLinkResult};
use crate::services::deep_link_service;
use crate::state::AppState;

#[tauri::command]
pub fn open_deep_link(
    state: State<'_, AppState>,
    input: OpenDeepLinkInput,
) -> Result<OpenDeepLinkResult, String> {
    deep_link_service::open_deep_link(&state, input)
}
